portfolio/src/main/webapp/script.js

```diff
@@ -58,11 +58,16 @@ function handleResponse(response) {
 function addTextToDom(text) {
     console.log("Adding text to DOM: "+text);
     const table = document.getElementById("comment-section");
-    const parsedText = JSON.parse(text);
+    const parsedText = JSON.parse(text); // this is the object with properties numComments and comments
+    parsedText.comments = JSON.parse(parsedText.comments);
```

Comment:
Is this parsing of the inner value necessary? If so, is it possible to make to make the response in a way that it doesn't need to be parsed multiple times?